### ISSUE TITLE
Raspberry Pi running Debian | Unable to launch grocy docker container

### ISSUE THREAD

**Issue by Fictitious-Rotor** [grocy/grocy-docker#152]
Hello! I'm running into this issue when trying to get the docker container running...

After typing
```bash
pi@wifipi:~/Grocy/grocy-docker$ export GROCY_IMAGE_TAG=v3.1.2-0
docker-compose pull
docker-compose up
```

I get
```bash
Building backend
Step 1/24 : ARG PLATFORM
Step 2/24 : FROM --platform=${PLATFORM} docker.io/alpine:3.14.2
 ---> 14119a10abf4
Step 3/24 : LABEL maintainer "Talmai Oliveira <to@talm.ai>, James Addison <jay@jp-hosting.net>"
 ---> Using cache
 ---> dee1ee28a14c
Step 4/24 : ARG GROCY_VERSION
 ---> Using cache
 ---> 7f1b7351a28c
Step 5/24 : ARG COMPOSER_VERSION
 ---> Using cache
 ---> db780126e440
Step 6/24 : ARG COMPOSER_CHECKSUM
 ---> Using cache
 ---> 3ba22d5146c7
Step 7/24 : RUN set -eux;        adduser -u 82 -D -S -G www-data www-data
 ---> [Warning] The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
 ---> Running in cbf69727d5e2
standard_init_linux.go:219: exec user process caused: exec format error
ERROR: Service 'backend' failed to build: The command
```

Looking at the error message and poking through `docker-compose.yml`
```yaml
backend:
    image: "grocy/backend:${GROCY_IMAGE_TAG}"
    build:
      args:
        GROCY_VERSION: v3.1.2
        PLATFORM: linux/amd64 # <-- !!
```
I can see that this build won't be accommodating for for ARM.
Are there any other versions that I can use, or will it not be possible for me to use grocy here?

**Comment by jayaddison** [comment]
Hey @Fictitious-Rotor - thanks for trying this out.  If you modify the `PLATFORM` argument to `linux/arm64/v8`, are you able to build and run the containers?

**Comment by Fictitious-Rotor** [comment]
```
version: '2.4'

services:

  frontend:
    image: "grocy/frontend:${GROCY_IMAGE_TAG}"
    build:
      args:
        GROCY_VERSION: v3.1.2
        PLATFORM: linux/arm64/v8
      context: .
      dockerfile: Dockerfile-grocy-frontend
    depends_on:
      - backend
    ports:
      - '127.0.0.1:80:8080'
      - '127.0.0.1:443:8443'
    read_only: true
    tmpfs:
      - /tmp
    volumes:
      - /var/log/nginx

  backend:
    image: "grocy/backend:${GROCY_IMAGE_TAG}"
    build:
      args:
        GROCY_VERSION: v3.1.2
        PLATFORM: linux/arm64/v8
        COMPOSER_VERSION: "2.1.5"
        COMPOSER_CHECKSUM: "be95557cc36eeb82da0f4340a469bad56b57f742d2891892dcb2f8b0179790ec"
      context: .
      dockerfile: Dockerfile-grocy-backend
    expose:
      - '9000'
    read_only: true
    tmpfs:
      - /tmp
    volumes:
      - /var/log/php8
      - app-db:/var/www/data
    env_file:
      - grocy.env

volumes:
  app-db:
```

I've updated both references to PLATFORM, then reran docker-compose pull & up

```
pi@wifipi:~/Grocy/grocy-docker$ docker-compose up  
Creating network "grocy-docker_default" with the default driver
Building backend
Step 1/24 : ARG PLATFORM
Step 2/24 : FROM --platform=${PLATFORM} docker.io/alpine:3.14.2
 ---> 14119a10abf4
Step 3/24 : LABEL maintainer "Talmai Oliveira <to@talm.ai>, James Addison <jay@jp-hosting.net>"
 ---> Using cache
 ---> dee1ee28a14c
Step 4/24 : ARG GROCY_VERSION
 ---> Using cache
 ---> 7f1b7351a28c
Step 5/24 : ARG COMPOSER_VERSION
 ---> Using cache
 ---> db780126e440
Step 6/24 : ARG COMPOSER_CHECKSUM
 ---> Using cache
 ---> 3ba22d5146c7
Step 7/24 : RUN set -eux;        adduser -u 82 -D -S -G www-data www-data
 ---> [Warning] The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
 ---> Running in 567641e40f23
standard_init_linux.go:219: exec user process caused: exec format error
ERROR: Service 'backend' failed to build: The command '/bin/sh -c set -eux;        adduser -u 82 -D -S -G www-data www-data' returned a non-zero code: 1
```
I still get the same issue. I don't understand where it's getting linux/amd64 from. Am I missing something here?

**Comment by pwltr** [comment]
@Fictitious-Rotor
Hello, you could try this container image from linuxserver. Works for me!
https://hub.docker.com/r/linuxserver/grocy

**Comment by jayaddison** [comment]
@Fictitious-Rotor I think the problem is that vanilla `docker` build step (called by `docker-compose up`, when necessary) doesn't provide much (any?) support for multi-platform builds.

I haven't tried this yet, but you might be able to install `qemu` and then use `docker buildx build` to perform a multi-platform build.  The Docker Blog has a good explanatory post with more details: https://www.docker.com/blog/multi-platform-docker-builds/

If you get time to explore that, please let us know - it'd be nice to improve the documentation and support for multi-platform `grocy-docker`.

**Comment by jayaddison** [comment]
This should be resolved by #156 - the container images deployed to Docker Hub are now multi-architecture, including support for Arm architectures.

Building locally (`docker-compose build`) may still require `qemu` for cross-platform builds.
